goldmane/pkg/aggregator/aggregator.go

```diff
@@ -472,42 +426,83 @@ func (a *LogAggregator) queryFlows(req *proto.FlowListRequest) *listResponse {
-		return &listResponse{nil, err}
+		return &listResponse{0, nil, err}
 	}
 
-	// If a sort order was requested, use the corersponding index to find the matching flows.
+	// If a sort order was requested, use the corresponding index to find the matching flows.
 	if len(req.SortBy) > 0 && req.SortBy[0].SortBy != proto.SortBy_Time {
 		if idx, ok := a.indices[req.SortBy[0].SortBy]; ok {
 			// If a sort order was requested, use the corresponding index to find the matching flows.
 			// We need to convert the FlowKey to a string for the index lookup.
-			flows := idx.List(IndexFindOpts{
+			flows, totalPages := idx.List(IndexFindOpts{
 				startTimeGt: req.StartTimeGte,
 				startTimeLt: req.StartTimeLt,
-				limit:       req.PageSize,
-				page:        req.PageNumber,
+				pageSize:    req.PageSize,
+				page:        req.Page,
 				filter:      req.Filter,
 			})
 
 			// Convert the flows to proto format.
 			flowsToReturn := a.flowsToResult(flows)
-			return &listResponse{flowsToReturn, nil}
+			return &listResponse{int64(totalPages), flowsToReturn, nil}
 		} else if !ok {
-			return &listResponse{nil, fmt.Errorf("unsupported sort order")}
+			return &listResponse{0, nil, fmt.Errorf("unsupported sort order")}
 		}
 	}
 
 	// Default to time-sorted flow data.
-	flows := a.defaultIndex.List(IndexFindOpts{
+	flows, totalPages := a.defaultIndex.List(IndexFindOpts{
 		startTimeGt: req.StartTimeGte,
 		startTimeLt: req.StartTimeLt,
-		limit:       req.PageSize,
-		page:        req.PageNumber,
+		pageSize:    req.PageSize,
+		page:        req.Page,
 		filter:      req.Filter,
 	})
 
 	// Convert the flows to proto format.
 	flowsToReturn := a.flowsToResult(flows)
-	return &listResponse{flowsToReturn, nil}
+	return &listResponse{int64(totalPages), flowsToReturn, nil}
+}
+
+func (a *LogAggregator) queryFilterHints(req *proto.FilterHintsRequest) *filterHintsResponse {
+	logrus.WithFields(logrus.Fields{"req": req}).Debug("Received flow request")
+
+	// Sanitize the time range, resolving any relative time values.
+	req.StartTimeGte, req.StartTimeLt = a.normalizeTimeRange(req.StartTimeGte, req.StartTimeLt)
+
+	// Validate the request.
+	if err := a.validateTimeRange(req.StartTimeGte, req.StartTimeLt); err != nil {
+		return &filterHintsResponse{0, nil, err}
+	}
+
+	var sortBy proto.SortBy
+	switch req.Type {
+	case proto.FilterType_FilterTypeDestName:
+		sortBy = proto.SortBy_DestName
+	case proto.FilterType_FilterTypeDestNamespace:
+		sortBy = proto.SortBy_DestNamespace
+	case proto.FilterType_FilterTypeSourceName:
+		sortBy = proto.SortBy_SourceName
+	case proto.FilterType_FilterTypeSourceNamespace:
+		sortBy = proto.SortBy_SourceNamespace
+	}
+
+	// If a sort order was requested, use the corresponding index to find the matching flows.
+	if idx, ok := a.indices[sortBy]; ok {
+		// If a sort order was requested, use the corresponding index to find the matching flows.
+		// We need to convert the FlowKey to a string for the index lookup.
```

Comment:
```suggestion
```

Reply:
@Brian-McM I think this one still doesn't apply - we're not returning "Flows" here and we're not converting FLowKeys?